### DISCUSSION TITLE
Store/Provider not cleaning atoms on remounting

### DISCUSSION THREAD

**Discussion by omertavor365** [pmndrs/jotai#2423] [Q&A] [answered]
Hi,

I have a `MainControlView` component that holds MUI Tabs for each section in my view/page :

```
const MainControlView: React.FC = () => {
  const { currentTab, handleCurrentTabChange } = useMainControlView();

  return (
    <div style={styles.controlViewContainer}>
      <TabContext value={currentTab}>
        <Box sx={styles.tabsListContainer}>
          <TabList onChange={handleCurrentTabChange}>
            <Tab label="Countries" value={TERMS_TABS_IDS.COUNTRIES} />
            <Tab label="Competitions" value={TERMS_TABS_IDS.COMPETITIONS} />
            <Tab label="Competitors" value={TERMS_TABS_IDS.COMPETITORS} />
          </TabList>
        </Box>
        <TabPanel value={TERMS_TABS_IDS.COUNTRIES} sx={styles.tabPanel}>
          <CountriesArea />
        </TabPanel>
        <TabPanel value={TERMS_TABS_IDS.COMPETITIONS} sx={styles.tabPanel}>
          <CompetitionsArea />
        </TabPanel>
        <TabPanel value={TERMS_TABS_IDS.COMPETITORS} sx={styles.tabPanel}>
          <CompetitorsArea />
        </TabPanel>
      </TabContext>
    </div>
  );
};
```

Now, each children component under the TabPanels ( `CountriesArea` / `CompetitionsArea` / `CompetitorsArea` ) is being remounted when the fitting tab is selected (and not mounted when the tab is not selected).

Each one of the  `CountriesArea` / `CompetitionsArea` / `CompetitorsArea`  components has its own Jotai Store and Provider.

This is the `CountriesArea` component for reference (relevant stuff only):
(*The other 2 "Area" components look like that as well)

```
const countriesContextStore = createStore();

const CountriesArea: React.FC = () => {
  useEffect(() => {
    console.log('this logs only on mount');

    return () => {
      console.log('this logs only on UNmount');
    };
  }, []);

  return (
    <Provider key={'CountriesAreaID'} store={countriesContextStore}>
      {/* {inner content...} */}
    </Provider>
  );
};
```

In conclusion, each "Area" component has its own Store/Provider, and they are being remounted by the Tabs mechanism, so I figured the atoms under each Provider will be cleaned up on each remount - **but that doesn't seem to be the case, the atoms values are kept between remounts and not being cleaned up.**

I'll appreciate any help on what is missing/any solutions to the problem.
Thank you !

**Comment by rothsandro** [comment, marked as answer]
The `Provider` is just there to provide a store instance to all children, the actual atom values are kept in the store instance. You create the store instance outside of the component, so it will be kept when unmounting the component and reused when mounting the component again.

Do you use `countriesContextStore` somewhere else? If no, delete it and don't pass a store to the provider. The provider will create its own store which will be destroyed on unmount.

**Reply by omertavor365** [reply]
Thanks, It worked !

Your suggestion make sense, I even thought earlier to create `countriesContextStore` inside the Area component but it didn't seem like the right practice from reading the docs, but now I guess Provider does that by default.

The info you gave me in the answer is important, maybe it should be added the the Provider page in the Jotai docs, I'm sure many would benefit from that (obviously it's just a suggestion from my part).
